WordPress/Classes/ViewRelated/Feature Highlight/Tooltip.swift

```diff
@@ -251,32 +272,72 @@ final class Tooltip: UIView {
+    }
+
     private static func height(
         withTitle title: String?,
-        message: String?) -> CGFloat {
-            var totalHeight: CGFloat = 0
+        message: String?
+    ) -> CGFloat {
+        var totalHeight: CGFloat = 0
 
-            totalHeight += Constants.Spacing.contentStackViewTop
+        totalHeight += Constants.Spacing.contentStackViewTop
 
-            if let title = title {
-                totalHeight += title.height(withMaxWidth: Constants.maxWidth, font: Constants.Font.title)
-            }
+        if let title = title {
+            totalHeight += title.height(withMaxWidth: Constants.maxContentWidth, font: Constants.Font.title)
+        }
 
-            totalHeight += Constants.Spacing.contentStackViewInterItemSpacing
+        totalHeight += Constants.Spacing.contentStackViewInterItemSpacing
 
-            if let message = message {
-                totalHeight += message.height(withMaxWidth: Constants.maxWidth, font: Constants.Font.message)
-            }
+        if let message = message {
+            totalHeight += message.height(withMaxWidth: Constants.maxContentWidth, font: Constants.Font.message)
+        }
+
+        totalHeight += Constants.Spacing.buttonStackViewHeight
+        totalHeight += Constants.Spacing.contentStackViewBottom
+
+        return totalHeight
+    }
 
-            totalHeight += Constants.Spacing.buttonStackViewHeight
-            totalHeight += Constants.Spacing.contentStackViewBottom
+    private static func width(
+        title: String?,
+        message: String?,
+        primaryButtonTitle: String?,
+        secondaryButtonTitle: String?
+    ) -> CGFloat {
 
-            return totalHeight
+        let titleWidth: CGFloat
+        if let title = title {
+            titleWidth = title.width(withMaxWidth: Constants.maxContentWidth, font: Constants.Font.title)
+        } else {
+            titleWidth = 0
+        }
+
+        let messageWidth: CGFloat
+        if let message = message {
+            messageWidth = message.width(withMaxWidth: Constants.maxContentWidth, font: Constants.Font.message)
+        } else {
+            messageWidth = 0
+        }
+
+        var buttonsWidth: CGFloat = 0
+        if let primaryButtonTitle = primaryButtonTitle {
+            buttonsWidth += primaryButtonTitle.width(withMaxWidth: Constants.maxContentWidth, font: Constants.Font.button)
+        }
+
+        if let secondaryButtonTitle = secondaryButtonTitle {
+            buttonsWidth += secondaryButtonTitle.width(withMaxWidth: Constants.maxContentWidth, font: Constants.Font.button) + Constants.Spacing.buttonsStackViewInterItemSpacing
         }
+
+        return max(max(titleWidth, messageWidth), buttonsWidth) + Constants.Spacing.contentStackViewHorizontal * 2
+    }
 }
 
 private extension String {
-    func height(withMaxWidth maxWidth: CGFloat, font: UIFont) -> CGFloat {
+    private func size(withMaxWidth maxWidth: CGFloat, font: UIFont) -> CGRect {
```

Comment:
Since this method is already within a private extension, the `private` keyword can be removed.
```suggestion
    func size(withMaxWidth maxWidth: CGFloat, font: UIFont) -> CGRect {
```

Reply:
The goal here was to hide the `height` function from the rest of the code. Notice `private extension` and `private func` are at different levels of hierarchy. So while the extension can still be accessed by the class in the same file, the `height` function is inaccessible.